packages/server/src/routes/members/flag.ts

```diff
@@ -0,0 +1,200 @@
+import {
+  CollectionReference,
+  Firestore,
+  Transaction,
+  DocumentSnapshot
+} from "@google-cloud/firestore";
+import { firestore } from "firebase-admin";
+
+import {
+  OperationType,
+  Operation
+} from "@raha/api-shared/dist/models/Operation";
+import { OperationId } from "@raha/api-shared/dist/models/identifiers";
+import {
+  FlagMemberApiEndpoint,
+  ResolveFlagMemberApiEndpoint
+} from "@raha/api-shared/dist/routes/members/definitions";
+import { NotFoundError } from "@raha/api-shared/dist/errors/RahaApiError/NotFoundError";
+import { MemberIsFlaggedError } from "@raha/api-shared/dist/errors/RahaApiError/members/flag/MemberIsFlagged";
+import { MemberVerificationLevelTooLowError } from "@raha/api-shared/dist/errors/RahaApiError/members/flag/MemberVerificationLevelTooLow";
+
+import { OperationToInsert, createApiRoute } from "..";
+
+/**
+ * Right now "good standing" is somewhat arbitrarily defined as:
+ *  * verified by 5+ people
+ *  * not themselves flagged
+ */
+async function _validateMemberStanding(
+  transaction: Transaction,
+  member: DocumentSnapshot,
+  operationsCollection: CollectionReference
+) {
+  const operationsFlaggingMember: OperationId[] | undefined = member.get(
+    "operationsFlaggingThisMember"
+  );
+  if (operationsFlaggingMember && operationsFlaggingMember.length > 0) {
+    throw new MemberIsFlaggedError();
```

Comment:
Yeah it's a bit tedious to create new errors for every error type lol. That being said, I do appreciate exiting control flow whether we do that by throwing or returning upon an error, since I'm pretty sure in the past I've found it gets complicated to manage state if you try to keep a running tally of errors. Meh, yeah, maybe part of a refactor later.

Reply:
i think it's a generalizable problem that's probably been solved many times, probably a way to make it pretty simple but save it for later